src/resolve/resolved-varlink.c

```diff
@@ -582,7 +582,7 @@ static int varlink_notification_server_init(Manager *m) {
         if (r < 0)
                 return log_error_errno(r, "Failed to register varlink disconnect handler: %m");
 
-        r = varlink_server_listen_address(server, "/run/systemd/resolve/io.systemd.Resolve.Monitor", 0660);
+        r = varlink_server_listen_address(server, "/run/systemd/resolve/io.systemd.Resolve.Monitor", 0600);
```

Comment:
@poettering - This seems a bit restrictive - I wonder if we could leave this at 0660?

One use case of enabling group permissions is to have another systemd service monitor notifications when using "SupplementaryGroups=" to run in the same group as systemd-resolved.

Reply:
We shouldn't overload the group resolved itself runs in as a concept of allowing access to monitoring. These are two distinct concepts.  Moreover, this doesn't even work at all, since we also insist on the client being privileged because we set VARLINK_SERVER_ROOT_ONLY for the varlink server, i.e. we check the socket peer's UID already, so it won't help you if you are a member of the group. This is hence mostly just an extra level of security, nothing more, and enforces what we enforce in code anyway.

If group-based access to monitoring should be allowed it hence needs explicitly support from our side, and it should be done via a group that is solely about monitoring, and doesn't overload this otherwise. Because the "systemd-resolved" group as it stands now is a lot more than just access to the monitor socket, it in fact allows you to impersonate systemd-resolved to some level, and we should not allow that or make that officially supported.

Reply:
Ok thanks for clarifying!